compute_tools/src/compute.rs

```diff
@@ -1386,6 +1387,27 @@ LIMIT 100",
         }
         Ok(remote_ext_metrics)
     }
+
+    /// Waits until current thread receives a state changed notification and
+    /// the pageserver connection strings has changed.
+    ///
+    /// The operation will time out after a specified duration.
+    pub fn wait_timeout_while_pageserver_connstr_unchanged(&self, duration: Duration) {
+        let state = self.state.lock().unwrap();
+        let old_pageserver_connstr = state.pspec.as_ref().unwrap().pageserver_connstr.clone();
```

Comment:
Elsewhere, we use an `expect("pspec must be set")` which is a little bit easier to read than a plain unwrap

Reply:
improve readability: 4d14b41c876694c73d7bacd50150f76330a41f83